src/connection.js

```diff
@@ -1533,6 +1616,39 @@ Connection.prototype.STATE = {
       }
     }
   },
+  SENT_LOGIN7_WITH_KERBEROS: {
```

Comment:
The state machine is starting to get really complicated. Do you think kerberos authentication could be implemented without adding a new state? I know that there is also a special state for NTLM authentication defined, but while working on #624, I was thinking if we maybe could get rid of authentication specific states somehow. 🤔